src/react-app-env.d.ts

```diff
@@ -1 +1,24 @@
 /// <reference types="react-scripts" />
+
+interface User {
+  id: number;
+  name: string;
+  email: string;
+}
+
+interface Posts {
+  userId: number;
+  id: number;
+  title: string;
+  body: string;
+  user: User | null;
+  comments: Comments[] | null;
+}
+
+interface Comments {
+  postId: number;
```

Comment:
Ты описываешь один объект
```suggestion
interface Post {
  userId: number;
  id: number;
  title: string;
  body: string;
  user: User | null;
  comments: Comments[] | null;
}

interface Comment {
  postId: number;
```

Reply:
TS не нравилось почему-то Comment without 's', заменил на Commentary